corePlugins/margin.js

```diff
@@ -10,4 +10,4 @@ export default generator.generate('m', 'margin', theme.margin, [
     ['l', 'marginLeft'],
     ['s', 'marginStart'],
     ['e', 'marginEnd'],
-]);
+], '');
```

Comment:
I was wondering why this empty string is added 🤔

Reply:
This was added to normalize separators in the generator. Let me say this with example.
```js
import generator from '../util/generator';
 import theme from '../util/configHandler';

 export default generator.generate('m', 'margin', theme.margin, [
     ['x', 'marginHorizontal'],
     ['y', 'marginVertical'],
     ['t', 'marginTop'],
     ['r', 'marginRight'],
     ['b', 'marginBottom'],
     ['l', 'marginLeft'],
     ['s', 'marginStart'],
     ['e', 'marginEnd'],
 ]);
```

The generator would make its variant by using a separator ('-').
```js
for (; j < variationLength; ++j) {
                 i = 0;
                 keyName = `${name}-${variation[j][0]}`;
                 keyName = `${name}${variantSpacing}${variation[j][0]}`;
                 keys = variation[j][1];
```

Then those strings were converted to camelCase objects. So, this is great when it comes to variants like `display-none` and others. Which is converted to `displayNone`. 
When it comes to the variant of padding it makes this like `p-x` which is converted to `pX`. But here no separator was intended. So bu using custom variant separator it solves this as that is converted to `px` which is converted to `px.

These custom separators only used in padding and margin. To normalize it.

Reply:
I understand now, thank you for the explanation😊